packages/react-radio/src/components/Radio/Radio.types.ts

```diff
@@ -52,6 +53,24 @@ export type RadioProps = Omit<ComponentProps<Partial<RadioSlots>, 'input'>, 'siz
    * Disable this Radio item.
    */
   disabled?: boolean;
+
+  /**
+   * Callback when this Radio is selected in its group.
+   *
+   * **Note:** `onChange` is NOT called when this Radio is deselected.
+   * Use RadioGroup's `onChange` event to determine when the selection in the group changes.
+   */
+  onChange?: (ev: React.FormEvent<HTMLInputElement>, data: RadioOnChangeData) => void;
```

Comment:
Should this use a `ChangeEvent` instead of a `FormEvent`?

```suggestion
  onChange?: (ev: React.FormEvent<HTMLInputElement>, data: RadioOnChangeData) => void;
```
```suggestion
  onChange?: (ev: React.ChangeEvent<HTMLInputElement>, data: RadioOnChangeData) => void;
```

Reply:
Actually, I see why in the description for #22065 (maybe that comment should go here instead). Dismissing this comment.

Reply:
I think you're right that it should be changed to `ChangeEvent` for `Radio` just not `RadioGroup`.